### PR TITLE
Add Alert Component

### PR DESCRIPTION
**Pull request by chhavijustme**
Hello, 
Tried passing children as you suggested, hope I got it right. Let me know your feedback on this one :)

### CODE REVIEW

File: components/Alert.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  handleDismiss() {
+    this.setState({show: false});
+  }
+
+  render() {
+    if (this.state.show) {
+      return (
+        <Alert bsStyle={this.props.bsStyle} onDismiss={this.handleDismiss}>
+
```
Comment (samj1912) [inline components/Alert.js]: Extra WS?

File: stories/index.stories.js [hunk 2]
```diff
@@ -1,19 +1,60 @@
...
+import AlertDismissable from '../components/Alert';
 
 import '../less/styles.less';
 
 // Stories of components
+storiesOf('Alert', module)
+  .add('Danger Alert', withInfo('')(() => (
+    <div>
+      <AlertDismissable bsStyle="danger">
+        <h4> {'Delete this entity'} </h4>
```
Comment (mwiencek) [inline stories/index.stories.js]: I wonder about this example. My understanding is that alerts are for information/feedback on an action after it's taken, not for prompts that initiate actions. So I'd think a modal would be more suited for this example. What do you think?

File: components/Alert.js (hunk 1 above)
Comment (mwiencek) [reply components/Alert.js]: Fixed in b8972cd

File: stories/index.stories.js (hunk 2 above)
Comment (chhavijustme) [reply stories/index.stories.js]: Now that you put it that way, it does not make sense to have that example :) Updated the examples, couldn't think of a better one for Danger alert. What do you think will be more apt as an example here?!

Comment (mwiencek) [reply stories/index.stories.js]: What you have now seems fine. :) It's really just an example to demonstrate behavior and we can always change it later.

**Comment by mwiencek** [comment]
I have a couple small issues when I look at this in storybook:

storybook1

In this screenshot, a "Show Info" button (added by storybook) is blocking the "X" on the component. Can we add some margin around the alert so that the X is always visible?

storybook2

In this screenshot, I removed the "Show Info" button from the DOM so that the X is visible, but the X appears outside the alert. It could just be a glitch from me hand-editing things, but could also be an actual style bug (we'll see if it still gets placed outside the alert after adding some margin).

**Comment by chhavijustme** [comment]
Hey, I added a decorator for all stories to add a margin. It looks pretty fine! I am just not sure if we are to keep that decorator in the index.stories file or move it somewhere and import it? :)

File: stories/index.stories.js [hunk 3]
```diff
@@ -1,19 +1,64 @@
...
+  </div>
+);
+addDecorator(MarginDecorator);
+
 // Stories of components
+storiesOf('Alert', module)
+  .add('Danger Alert', withInfo('')(() => (
+    <div>
+      <AlertDismissable bsStyle="danger">
+        <p>{'You just deleted the entitiy. The changes will now be voted on. Click '}<a className="alert-link">{'here'}</a> {'to know more!'}</p>
```
Comment (mwiencek) [inline stories/index.stories.js]: typo: s/entitiy/entity/

File: stories/index.stories.js [hunk 4]
```diff
@@ -1,19 +1,64 @@
...
+      </AlertDismissable>
+    </div>
+  )))
+
+  .add('Info Alert', withInfo('')(() => (
+    <div>
+      <AlertDismissable bsStyle="info">
+        <p>{'Hmm, you cannot edit it. Permissions to delete this entity are only granted to priviledged editors. Click'} <a className="alert-link">{'here'}</a> {'to know more!'} </p>
```
Comment (mwiencek) [inline stories/index.stories.js]: typo: s/priviledged/privileged/

File: less/alerts.less [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  }
+
+  > p + p {
+    margin-top: 5px;
+  }
+}
+
+.alert-dismissable, // The misspelled .alert-dismissable was deprecated in 3.2.0.
+.alert-dismissible {
+  padding-right: (@alert-padding + 20);
```
Comment (mwiencek) [inline less/alerts.less]: Hm, this is correct, but it's unfortunate that we have to add duplicate statements all over (due to how CSS works).

I think we made a mistake by creating separate files with our changes. It'd be much cleaner to just modify the original bootstrap source. It's still easy to see the modifications and generate a patch with git. But let's not worry about that here.

**Comment by mwiencek** [comment]
> It looks pretty fine! I am just not sure if we are to keep that decorator in the index.stories file or move it somewhere and import it? :)

What you have looks perfect. 👍

**Comment by chhavijustme** [comment]
Hi, I increased the opacity of the close button to make it more clear. I made changes to the bootstrap less file itself, as it was just a one digit change. Let me know if I should rather make a new less file to avoid messing/causing confusion! :)

**Pull request merged**